src/P5Functions/Math.js

```diff
@@ -433,7 +433,14 @@ module.exports = function withMath(p, undef) {
       return internalRandomGenerator() * arguments[0];
     }
     var aMin = arguments[0], aMax = arguments[1];
-    return internalRandomGenerator() * (aMax - aMin) + aMin;
+    while (true) {
```

Comment:
slightly wary of a `while(true)`, in theory this *can* run forever, so having a cutoff count to ensure that this cannot yield an infinite loop would be wise.
```
var maxtries = 100;
while(maxtries--) {
 ...
}
```
Although that said, if we have numbers for which `p.random` is demonstrably yielding the wrong result, we might even be able to figure out why the `result` computation can end up being wrong, and fix it without needing the while loop here.